mllib/src/main/scala/org/apache/spark/mllib/clustering/KMeans.scala

```diff
@@ -287,6 +291,10 @@ class KMeans private (
 
       val bcActiveCenters = sc.broadcast(activeCenters)
 
+      if (instr != null) {
+        instr.logNumFeatures(bcActiveCenters.value(0)(0).vector.size)
```

Comment:
This is being logged on every iteration, but it should only be logged once.  Move before the while loop, and set it using "centers".